jwst/assign_wcs/nirspec.py

```diff
@@ -49,8 +49,11 @@ def create_pipeline(input_model, reference_files):
     reference_files : dict
         {reftype: reference_file_name} mapping.
     """
-    exp_type = input_model.meta.exposure.type.lower()
-    pipeline = exp_type2transform[exp_type](input_model, reference_files)
+    if input_model.meta.instrument.grating.lower() == "mirror":
+        pipeline = imaging(input_model, reference_files)
+    else:
+        exp_type = input_model.meta.exposure.type.lower()
+        pipeline = exp_type2transform[exp_type](input_model, reference_files)
     if pipeline:
         log.info("Created a NIRSPEC {0} pipeline with references {1}".format(
                 exp_type, reference_files))
```

Comment:
If the new line 52 is True, then line 55 won't be executed and hence "exp_type" won't yet be defined in line 59 (caught by Travis tests - not me).